spring-ai-core/src/main/java/org/springframework/ai/chat/ChatOptions.java

```diff
@@ -35,4 +38,12 @@ public interface ChatOptions extends ModelOptions {
 
 	void setTopK(Integer topK);
 
+	default List<ToolFunctionCallback> getToolCallbacks() {
+		throw new UnsupportedOperationException("ToolCallbacks is not supported by this model");
```

Comment:
I think we should not have this default method, since the methods here should apply to every model.